### PR TITLE
Add basic test setup

### PR DESCRIPTION
**Pull request by sir-marc**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,8 +8,15 @@
     "install-be": "cd ./game-backend && npm install",
     "build-fe": "cd ./game-frontend && cross-env NODE_ENV=production npm run build",
     "build-be": "cd ./game-backend && cross-env NODE_ENV=production npm run build",
+    "start-fe": "cd ./game-frontend && cross-env NODE_ENV=production npm run start",
+    "start-be": "cd ./game-backend && cross-env NODE_ENV=production npm run start",
     "test-fe": "cd ./game-frontend && cross-env NODE_ENV=test CI=true npm run test",
     "test-be": "cd ./game-backend && cross-env NODE_ENV=test CI=true npm run test",
+    "start-app": "npm run start-be && npm run start-fe",
```
Comment (StuckiSimon) [inline package.json]: Is this script working? The `start-be` won't exit and therefore `start-fe` won't be executed.

File: report/report.tex [hunk 2]
```diff
@@ -235,6 +235,25 @@ \subsection{CI /CD}
 Der Aufbau unserer Software hat in voraus bereits eine Grobplanung in welchen schritten und Phasen entwickelt wird.
 Dies ermöglicht es uns, bereits frühzeitig Fehler zu erkennen und Schnittstellen genauer zu spezifizieren.
 
+\subsection{Testing}
+Das Testen der Applikation erfolgt in drei Schritten:
+
+\subsubsection{Unit Testing}
+Unit Tests werden automatisiert bei jedem Build und Pull Request ausgeführt.
```
Comment (StuckiSimon) [inline report/report.tex:242]: Is this already setup?

File: report/report.tex [hunk 3]
```diff
@@ -235,6 +235,25 @@ \subsection{CI /CD}
...
+\subsubsection{Integration Testing}
+Integration Tests, die durchgeführt werden, während der Entwicklung und vor einem PR zum Überprüfen der Akzeptanzkriterien.
+Für die End-to-End-Tests wird Cypress verwendet. Auch hier ist damit bereits wertvolle Erfahrung im Team vorhanden
+und löste bislang immer Zufriedenheit bei der Benutzung aus.
+Integration Tests prüfen das Zusammenspiel der Applikation und überprüft die Gesamtfunktionalität.
+Die Applikation wird dabei aus der Sicht eines Benutzers automatisiert getestet und Testfälle basieren
+auf den Akzeptanzkriterien der User Stories.
+
+\subsubsection{Manuelles Testing}
+Manuelles Testen während der Entwicklungsphase und danach. Dieser Teil sollte so klein wie möglich gehalten werden, um die Qualität zu gewährleisten.
```
Comment (StuckiSimon) [inline report/report.tex]: How do we do manual regression testing? Is there a document describing what needs to be tested manually?

**Review by StuckiSimon — CHANGES_REQUESTED** [review]
Why did you commit the cypress sample files?
I'd prefer removing them.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+cypress/videos
```
Comment (StuckiSimon) [inline .gitignore]: What about adding a local .gitignore file in cypress?

File: package.json (hunk 1 above)
Comment (sir-marc) [reply package.json]: It is working, but don't ask me why 😅

File: report/report.tex (hunk 2 above)
Comment (sir-marc) [reply report/report.tex:242]: Jep

File: report/report.tex (hunk 3 above)
Comment (sir-marc) [reply report/report.tex]: Common sense ^^ Needs to be defined

File: .gitignore (hunk 4 above)
Comment (sir-marc) [reply .gitignore]: Maybe someday there will be more to be ignored

Comment (StuckiSimon) [reply .gitignore]: If it's in the root folder that's fine - but I'd go for the same convention we've already established for `report`, `game-frontend` as well as `game-backend`.

File: report/report.tex (hunk 2 above)
Comment (StuckiSimon) [reply report/report.tex:242]: Is it possible that the pipeline currently **deploys** for each PR? :v:
Please check the Build Pipeline logs in Azure...

File: report/report.tex (hunk 3 above)
Comment (StuckiSimon) [reply report/report.tex]: Maybe add a sentence about manual testing before creating a PR with the rebased changes from master? Doesn't have to be much, but at least something? Currently there is no information about when, how and who does manual testing.

File: package.json (hunk 1 above)
Comment (StuckiSimon) [reply package.json]: The `start-be` is working, but `start-fe` is not being executed on my machine.
Is it possible that you're running the frontend in another terminal session already?

File: cypress/integration/examples/actions.spec.js [hunk 5]
```diff
@@ -0,0 +1,298 @@
+/// <reference types="cypress" />
+
+context('Actions', () => {
```
Comment (StuckiSimon) [inline cypress/integration/examples/actions.spec.js]: Maybe I missed your reply on my feedback during the previous review, but why do we add all these sample tests into our project?

**Review by StuckiSimon — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (sir-marc) [reply package.json]: That was the case ^^
Fixed now

File: report/report.tex (hunk 2 above)
Comment (sir-marc) [reply report/report.tex:242]: It only builds on each PR, but does not deploy

File: .gitignore (hunk 4 above)
Comment (sir-marc) [reply .gitignore]: Done ✅

File: cypress/integration/examples/actions.spec.js (hunk 5 above)
Comment (sir-marc) [reply cypress/integration/examples/actions.spec.js]: I've added them as a reference, but could also delete them.

Comment (StuckiSimon) [reply cypress/integration/examples/actions.spec.js]: I'd advise to delete them as they serve no purpose.

File: cypress/fixtures/profile.json [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
+  "id": 8739,
+  "name": "Jane",
```
Comment (StuckiSimon) [inline cypress/fixtures/profile.json]: Even though I like Jane maybe deleting the test data as well would be nice?

File: cypress/integration/index.spec.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+/// <reference types="cypress" />
+
+context("Basic test", () => {
+  beforeEach(() => {
+    cy.visit("localhost:3000/");
```
Comment (StuckiSimon) [inline cypress/integration/index.spec.js]: Why not configure cypress baseurl?

**Review by StuckiSimon — CHANGES_REQUESTED** [review]

File: cypress/integration/index.spec.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
+/// <reference types="cypress" />
+
+context("Basic test", () => {
+  beforeEach(() => {
+    cy.visit("/");
+  });
+
+  it("should render react logo", () => {
+    // https://on.cypress.io/focus
```
Comment (StuckiSimon) [inline cypress/integration/index.spec.js]: @sir-marc what's the purpose of these comments?

**Review by StuckiSimon — COMMENTED** [review]
Please check the test for obsolete comments.

**Review by StuckiSimon — APPROVED** [review]
Solid.

**Pull request merged**